### PR TITLE
[backport: 1.2] Rename `=` to `=copy`

### PR DESCRIPTION
**Pull request by Clyybber**

### CODE REVIEW

File: compiler/sempass2.nim [hunk 1]
```diff
@@ -809,7 +809,8 @@ proc trackCall(tracked: PEffects; n: PNode) =
 
   if a.kind == nkSym and a.sym.name.s.len > 0 and a.sym.name.s[0] == '=' and
         tracked.owner.kind != skMacro:
-    let opKind = find(AttachedOpToStr, a.sym.name.s.normalize)
+    var opKind = find(AttachedOpToStr, a.sym.name.s.normalize)
+    if a.sym.name.s.normalize == "=": opKind = attachedAsgn.int
```
Comment (Araq) [inline compiler/sempass2.nim:813]: No need for ``normalize`` here.

**Comment by Araq** [comment]
Why do we need this commit:

Duplicate tests for =copy hook

**Comment by alaviss** [comment]
Isn't this a breaking change? Can we have some coverage on why is this change being made and why will it be backported to 1.2? I have projects that use the custom `=` hook.

**Comment by ghost** [comment]
@alaviss 
> Isn't this a breaking change? Can we have some coverage on why is this change being made and why will it be backported to 1.2? I have projects that use the custom `=` hook.

It's not a breaking change, old code with = will still work, but the new proposed way is =copy.

IIRC the justification for that change is that = name for a hook might be misleading, =copy is much cleaner

**Comment by alaviss** [comment]
That's good to know, but can we have this in the changelog? I don't think this rename has been covered much other than a few mentions here and there.

**Comment by Araq** [comment]
I took over, sorry. I really want to get 1.4 out.

**Pull request closed without merging**